src/components/CardPrimary/withTopBadge.tsx

```diff
@@ -0,0 +1,44 @@
+import { Badge } from "@/components/Badge/Badge";
+import { ComponentType, FC } from "react";
+import styled from "styled-components";
+
+export interface WithTopBadgeProps {
+  topBadgeText?: string;
+  isSelected?: boolean;
+}
+
+const Wrapper = styled.div`
+  position: relative;
+  width: 100%;
+  max-width: 100%;
+`;
+
+const TopBadge = styled(Badge)<{ $isSelected?: boolean }>`
+  position: absolute;
+  top: 0;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  ${({ $isSelected, theme }) =>
+    $isSelected
+      ? `border-color: ${theme.click.button.basic.color.primary.stroke.active};`
+      : ""}
+  div:active + & {
+    border-color: ${({ theme }) => theme.click.button.basic.color.primary.stroke.active};
+  }
+`;
+
+export const withTopBadge =
```

Comment:
initially i wanted to make this a decorator to any type of cards, but there are many little details that make it more complex than it needs to be right now (borders on active state which change between card types, fixed with vs 100% etc)

i have kept the decorator structure as it might be easier to export in the future